library/src/com/google/maps/android/PolyUtil.java

```diff
@@ -172,9 +172,37 @@ public static boolean isLocationOnPath(LatLng point, List<LatLng> polyline,
 
     private static boolean isLocationOnEdgeOrPath(LatLng point, List<LatLng> poly, boolean closed,
                                                   boolean geodesic, double toleranceEarth) {
+        int idx = locationIndexOnEdgeOrPath(point, poly, closed, geodesic, toleranceEarth);
+
+        return (idx >= 0);
+    }
+
+    /**
+     * Computes whether (add where) the given point lies on or near a polyline, within a specified
+     * tolerance in meters. The polyline is composed of great circle segments if geodesic
+     * is true, and of Rhumb segments otherwise. The polyline is not closed -- the closing
+     * segment between the first point and the last point is not included.
+     */
+    public static int locationIndexOnPath(LatLng point, List<LatLng> polyline,
+                                           boolean geodesic, double tolerance) {
+        return locationIndexOnEdgeOrPath(point, polyline, false, geodesic, tolerance);
+    }
+
+    /**
+     * Same as {@link #locationIndexOnPath(LatLng, List, boolean, double)}
+     *
+     * with a default tolerance of 0.1 meters.
+     */
+    public static int locationIndexOnPath(LatLng point, List<LatLng> polyline,
+                                           boolean geodesic) {
+        return locationIndexOnPath(point, polyline, geodesic, DEFAULT_TOLERANCE);
+    }
+
+    private static int locationIndexOnEdgeOrPath(LatLng point, List<LatLng> poly, boolean closed,
```

Comment:
In my opinion, it would make more sense to return N if the point is between poly[n-1] and poly[0]. This would allow to distinguish between:
- point is between poly[n-2] and poly[n-1]
- point is between poly[n-1] and poly[0]